distribution/lib/Standard/Base/0.0.0-dev/src/Data/Vector/Extensions.enso

```diff
@@ -0,0 +1,55 @@
+import project.Any.Any
+import project.Data.Array.Array
+import project.Data.Vector.Vector
+import project.Error.Error
+import project.Errors.Illegal_Argument.Illegal_Argument
+import project.Runtime
+
+## GROUP Selections
+   Swaps the rows and columns of a matrix represented by a `Vector` of `Vector`s.
+
+   ! Error Conditions
+
+     - If the rowws (subvectors) do not all have the same length, an
+     `Illegal_Argument` error is raised.
+
+   > Example
+     Transpose a `Vector` of `Vectors`.`
+
+         "Hello, world!".reverse
+
+         matrix = [[0, 1, 2], [3, 4, 5], [6, 7, 8]]
+         # +---+---+---+
+         # | 0 | 1 | 2 |
+         # +---+---+---+
+         # | 3 | 4 | 5 |
+         # +---+---+---+
+         # | 6 | 7 | 8 |
+         # +---+---+---+
+
+         transposed = [[0, 3, 6], [1, 4, 7], [2, 5, 8]]
+         # +---+---+---+
+         # | 0 | 3 | 6 |
+         # +---+---+---+
+         # | 1 | 4 | 7 |
+         # +---+---+---+
+         # | 2 | 5 | 8 |
+         # +---+---+---+
+
+         matrix.transposed == transposed
+         # => True
+Vector.transpose : Vector (Vector Any) ! Illegal_Argument
+Vector.transpose self =
+    if self.is_empty then [] else
+        length = self.length
+        first_subvector_length = self.at 0 . length
+        builders = Vector.new first_subvector_length (_-> Vector.new_builder length)
+        result = self.map v->
+            if v.length != first_subvector_length then Error.throw (Illegal_Argument.Error "Transpose requires that all rows be the same length") else
+                v.map_with_index i-> x->
+                    builders.at i . append x
+        result.if_not_error <|
+            builders.map .to_vector
```

Comment:
We could do this using a proxy:
```
        length = self.length
        first_subvector_length = self.at 0 . length
        if v.any s-> s.length != first_subvector_length then Error.throw (Illegal_Argument.Error "Transpose requires that all rows be the same length") else
            inner i  = Vector.from_polyglot_array (Array_Proxy.new length j->((self.at j). at i))
            proxy = Array_Proxy.new first_subvector_length inner
            Vector.from_polyglot_array proxy
```

Reply:
Any reason not to have this as a normal method?

Reply:
Done. No reason not to move it into Vector.enso. Is that preferred? I figured since it was for something more specific than `Vector` maybe it should be elsewhere.